src/metrics.ts

```diff
@@ -188,7 +188,8 @@ export const MetricsProcessor = (
     }
 
     const metrics: Metrics = _summarize();
-    if (metrics) {
+
+    if (metrics && metrics.metricsData.length + metrics.targetData.length > 0) {
```

Comment:
Super minor and works the same way, but you can treat these as `truthy`/`falsey`.
```suggestion
    if (metrics && metrics.metricsData.length && metrics.targetData.length) {
```

Reply:
That is unless you do want to send if `metricsData` is empty, but `targetData` isn't. That that case, your could `||` them
```suggestion
    if (metrics && (metrics.metricsData.length || metrics.targetData.length)) {
```